### ISSUE TITLE
`++` operator is right associativity

### ISSUE THREAD

**Issue by bobzhang** [facebook/reason#2577]
Most languages have `+` as left associativity, this applies to OCaml as well.

**Comment by TheSpyder** [comment]
isn't this because OCaml `^` is right associative and that's what `++` translates to?

**Comment by bobzhang** [comment]
In my opinion this is a leaky abstraction, when I am using reason syntax why do I need remember that it maps to ^ in ocaml

**Comment by TheSpyder** [comment]
Sorry, I didn't mean to question why we would change it, just why it ended up this way.

**Comment by jordwalke** [comment]
Yes, this should be changed and it's one of the things I've been planning to do in the next breaking change. It has to be in a breaking change though and those have to be batched. Thanks for reminding me.

**Comment by jordwalke** [comment]
The original reason for it being left associative is not a good reason. It was so that when converting from OCaml code, you don't end up with some really funny looking output like `"((hi" ++ "bye") ++ "there")`.
